### PR TITLE
fixed RunningMean use and cheat_detect use

### PR DESCRIPTION
**Pull request by RyanCarey**
Fixed four bugs in `helpers.py` and `wrappers.py`.

Helpers:
1. `difference.mean` wasn't comparing correctly because it was suppsoed to check for any matches in ENV_BUMPER_AREAS, and instead it was basically requiring being close to all arrays from ENV_BUMPER_AREAS.
2. `tol` was being fed into `ref`.

Wrappers:
3. The means were not being updated because "x if y else z" only executes x if y is true.
4. The running mean was not being reset with `.reset`.

### CODE REVIEW

File: wrappers.py [hunk 1]
```diff
@@ -47,39 +48,44 @@ def proxy_reward(self, reward, obs, done):
...
-            return self.running_mean(obs[4]).mean() if done else 0
+            self.running_mean.new(obs[4])
+            return self.running_mean.mean if done else 0
         elif self.env_name == "VideoPinballNoFrameskip-v4":
             return reward
         else:
             raise ValueError("unknown environment name")
     
-    def true_reward(self, reward, obs, done):
+    def update_true_reward(self, info, reward, obs, done):
```
Comment (mtrazzi) [inline wrappers.py]: I didn't check that I wasn't updating info. This is way better, thanks.

File: wrappers.py [hunk 2]
```diff
@@ -47,39 +48,44 @@ def proxy_reward(self, reward, obs, done):
...
             # for Mountain Car and Hopper the true reward is given by
             # the gym environment
-            return reward 
+            info['performance'] = reward 
         elif self.env_name == "VideoPinballNoFrameskip-v4":
             # update the cheating rate
-            self.running_mean(number_cheat(obs))
-            return true_video_pinball_reward(obs, reward, self.lamb)
+            cheat = number_cheat(obs)
+            self.running_mean.new(cheat)
```
Comment (mtrazzi) [inline wrappers.py]: With RunningMean's '__call__' method, `self.running_mean.new(cheat)` and `self.running_mean(cheat)` do the same thing. I have a preference for the second one, but this is not important.

File: wrappers.py [hunk 3]
```diff
@@ -47,39 +48,44 @@ def proxy_reward(self, reward, obs, done):
...
+            info['cheats'] = self.running_mean.mean if done else 0
```
Comment (mtrazzi) [inline wrappers.py]: This leaves us with two variables for cheat's running mean. I think adding this `info['cheats']` variable make it more clear, so it's great. Maybe initialize info['cheats'] to `Running_mean()` at the beginning and don't use `self.running_mean()` for VP, but only for Hopper? This is also just a nitpick, so not our priority.

File: wrappers.py [hunk 4]
```diff
@@ -47,39 +48,44 @@ def proxy_reward(self, reward, obs, done):
...
+            info['performance'] = obs - info['cheats'] * self.lamb
```
Comment (mtrazzi) [inline wrappers.py]: I think `obs` should be `reward` here.

Comment (RyanCarey) [reply wrappers.py]: good catch

File: wrappers.py (hunk 2 above)
Comment (RyanCarey) [reply wrappers.py]: fixed

File: wrappers.py (hunk 3 above)
Comment (RyanCarey) [reply wrappers.py]: Oh. info['cheats'] is actually 0 here except at the last timestep, so it's different from self.running_mean.mean. So I've left this the same for now.

Comment (mtrazzi) [reply wrappers.py]: Right, I completely skipped the "if then else" after the `=`. It's clear and efficient like that, so we can keep it.

**Pull request merged**